services/metadata/src/metadata/services/avus.clj

```diff
@@ -135,5 +140,7 @@
     (let [avu-ids (set (map :id (:avus (when template-id
                                          (update-metadata-template-avus
                                            user-id data-id data-type template-id metadata)))))]
-      (persistence/remove-orphaned-avus data-id avu-ids))
-    (list-metadata-template-avus data-id)))
+      (persistence/remove-orphaned-avus data-id avu-ids)))
+  ;; this one will duplicate the one sent by update-metadata-template-avus but we want to catch any orphaned AVUs that were removed and make sure it's outside of a transaction
+  (amqp/publish-metadata-update user-id data-id)
```

Comment:
This will usually be a second message sent -- if we start having issues with too much reindexing we'll want to make it possible to disable sending a message within `update-metadata-template-avus` in this case. I couldn't find anything that let me know if I was already in a transaction, which might be nicest -- if it's inside a DB transaction, always skip sending the message and wait for the call which is outside of the transaction.

Of course, having something that'd let me just run a hook when the outermost transaction commits would still be nicest :)